src/main/java/com/google/devtools/build/lib/rules/python/PyCommon.java

```diff
@@ -158,16 +167,42 @@ public void addCommonTransitiveInfoProviders(RuleConfiguredTargetBuilder builder
    *     addSkylarkTransitiveInfo(PYTHON_SKYLARK_PROVIDER_NAME, createSourceProvider(...))
    */
   public static SkylarkClassObject createSourceProvider(
-      NestedSet<Artifact> transitivePythonSources, boolean isUsingSharedLibrary) {
+      NestedSet<Artifact> transitivePythonSources,
+      boolean isUsingSharedLibrary,
+      PathFragment packageFragment,
+      NestedSet<PathFragment> imports) {
     return NativeClassObjectConstructor.STRUCT.create(
         ImmutableMap.<String, Object>of(
             TRANSITIVE_PYTHON_SRCS,
             SkylarkNestedSet.of(Artifact.class, transitivePythonSources),
             IS_USING_SHARED_LIBRARY,
-            isUsingSharedLibrary),
+            isUsingSharedLibrary,
+            PACKAGE_FRAGMENT,
```

Comment:
Given that this is computable from the package fragment, how about not having this function here? If needed, we could add a method to `Label` that tells its runfiles path, right, @laurentlb ?